src/platform/nxp/rt/rt1170/BUILD.gn

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) 2021 Project CHIP Authors
+# Copyright 2023 NXP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+import("//build_overrides/nxp_sdk.gni")
+import("//build_overrides/openthread.gni")
+import("${chip_root}/src/platform/device.gni")
+import("${nxp_sdk_build_root}/${nxp_sdk_name}/${nxp_sdk_name}.gni")
+import("${nxp_sdk_build_root}/nxp_sdk.gni")
+
+assert(chip_device_platform == "nxp")
+
+config("nxp_platform_config") {
+  defines = [ "EXTERNAL_BLEMANAGERIMPL_HEADER=\"platform/nxp/common/ble_zephyr/BLEManagerImpl.h\"" ]
+  include_dirs = [
+    ".",
+    "../../common",
+  ]
+  if (chip_with_factory_data == 1) {
```

Comment:
This seems to be a bug in how NXP defines values: gn is aware of 'bool' values and this should be set to tryue/false instead of 0/1. Please create a separate PR to fix this and update `third_party/nxp/nx_sdk.gni` and related files.